pkg/apis/kubermatic/v1/cluster.go

```diff
@@ -421,7 +421,7 @@ const (
 	ClusterConditionAddonControllerReconcilingSuccess                   ClusterConditionType = "AddonControllerReconciledSuccessfully"
 	ClusterConditionAddonInstallerControllerReconcilingSuccess          ClusterConditionType = "AddonInstallerControllerReconciledSuccessfully"
 	ClusterConditionBackupControllerReconcilingSuccess                  ClusterConditionType = "BackupControllerReconciledSuccessfully"
-	ClusterConditionCloudControllerReconcilingSuccess                   ClusterConditionType = "CloudControllerReconcilledSuccessfully"
+	ClusterConditionCloudControllerReconcilingSuccess                   ClusterConditionType = "CloudControllerReconciledSuccessfully"
```

Comment:
Aren't we going to need some sort of migration for this? What's going to happen with existing clusters when you upgrade?

Reply:
I added the misspelling back to the ENUM so that the CRD stays compatible. The cloud controller will now during its reconciliation simply set the new and remove the old condition, so that in KKP 2.22 we can remove the misspelling from the ENUM.

This will lead to a short moment in which Cluster objects seemingly do not have all necessary conditions to be considered "healthy", but this will sort itself out quickly.